Anature/src/application/controllers/BattleController.java

```diff
@@ -435,9 +482,32 @@ private void setUpSwitchElements(Scene scene)
 		createBindsImageView(mSwitchDialogue, scene, 1, 1, mShowSwitch);
 
 		createBindsImageView(mSwitchBtn, scene, 2.71, 1.5, 8.31, 21.818, mShowSwitch);
-		mSwitchBtn.setOnMouseClicked(event -> activateTurn(BattleChoice.Switch));
+		mSwitchBtn.setOnMouseClicked(event ->
+		{
+			if(mShowSwitchBackBtn.get())
+			{
+				activateTurn(BattleChoice.Switch);
+			}
+			else
+			{
+				activateSwitch();
```

Comment:
Need to add a health check before this line so that the player can't switch to a dead Anature.
Soemthing like this:
if(mPlayer.getAnatures().get(mSwitchIndexSelected).getCurrHp() <= 0)
{
return;
}